components/organisms/PortfolioIndicator.tsx

```diff
@@ -1,29 +1,153 @@
-import React from "react";
-import {View} from "../atoms/View";
-import {BorrowText, DepositText, SecondaryText, StandardText} from "../typography";
+import React, { useEffect, useState } from "react";
+import { BigNumber } from "bignumber.js";
+import { View } from "../atoms/View";
+import { BorrowText, DepositText, SecondaryText, StandardText } from "../typography";
+import { StyleSheet } from "react-native";
+import Svg, { ClipPath, G, Defs, LinearGradient, Stop, Circle, Polygon } from "react-native-svg";
 
 export const PortfolioIndicator = () => {
     // TODO -- get collateral, borrowed and rate
-    const collateral = "$8,824.21";
-    const rate = 2.14;
-    const borrowed = "$3,333.21";
-  return <View flexDirection={"row"} paddingVertical={20}>
-      <View flex={1} alignItems={"center"} justifyContent={"center"}>
-          <SecondaryText>
-              Collateral
-          </SecondaryText>
-          <DepositText>
-              {collateral}
-          </DepositText>
-      </View>
-      <View flex={1} alignItems={"center"} justifyContent={"center"}>
-          <StandardText>
-              {rate}
-          </StandardText>
-      </View>
-      <View flex={1} alignItems={"center"} justifyContent={"center"}>
-          <SecondaryText>Borrowed</SecondaryText>
-          <BorrowText>{borrowed}</BorrowText>
-      </View>
-  </View>;
+    const collateral = "6000";
+    const borrowed = "5000";
+
+    const sizeCircle = 100;
+    const radio = sizeCircle/2;
+    const strockeWidth = 20;
+    const styleCircle = {
+        height: sizeCircle,
+        width: sizeCircle,
+        borderRadius: sizeCircle/2,
+    }
+    const styleInsideCircle ={
+        height: sizeCircle - strockeWidth,
+        width: sizeCircle - strockeWidth,
+        borderRadius: (sizeCircle - strockeWidth)/2,
+    }
+
+    const collateralBN = new BigNumber(collateral);
+    const borrowedBN = new BigNumber(borrowed);
+    const rate = collateralBN.dividedBy(borrowedBN);
+    
+    const angulo = Math.PI/2 * (1 - parseFloat(rate.toString()));
+    const x = 2 * radio * Math.cos(angulo) + radio;
+    const y = radio - 2 * radio * Math.sin(angulo);
+
+    const sizeBall = sizeCircle/5;
+    const radioCenterBall = radio;
+    const xBall = radio + radioCenterBall * Math.cos(angulo) - sizeBall/2 - (strockeWidth/2 * Math.cos(angulo))/2;
+    const yBall = radio - radioCenterBall * Math.sin(angulo) - sizeBall/2 + (strockeWidth/2 * Math.sin(angulo))/2;
```

Comment:
Can we move everything related to calculating these values on a separated function pls